scripts/classification/imagenet/train_imagenet.py

```diff
@@ -122,6 +127,11 @@ def main():
 
     logger.info(opt)
 
+    assert not opt.auto_layout or opt.amp, "--auto-layout needs to be used with --amp"
+
+    if opt.amp:
+        amp.init(layout_optimization=opt.auto_layout)
```

Comment:
Referring to definition of `amp.init()` [here](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/contrib/amp/amp.py#L283), seems there is no argument like `layout_optimization` ?

Reply:
It's an internal feature, it will be added soon

Reply:
Thanks for clarification.

Reply:
Just curiously, when setting both `--amp` and `--dtype float16`, what will be happening?